corehq/project_limits/tests/test_rate_definition.py

```diff
@@ -151,19 +160,11 @@ def test_rate_definition_against_fixed_user_table():
         per_second=0.005,
     )
     floor_rate = RateDefinition(
+        per_week=100,
         per_day=50,
         per_hour=30,
         per_minute=10,
         per_second=1,
     )
     for n_users, per_second, per_minute, per_hour, per_day, per_week in table:
-        testil.eq(
-            per_user_rate.times(n_users).with_minimum(floor_rate).map(int),
-            RateDefinition(
-                per_week=per_week,
-                per_day=per_day,
-                per_hour=per_hour,
-                per_minute=per_minute,
-                per_second=per_second,
-            ),
-        )
+        yield check, n_users, per_second, per_minute, per_hour, per_day, per_week
```

Comment:
just for my own understanding, why is this kind of test better?

Reply:
Each yielded item is a separate test, which can pass or fail on its own. It is similar to (and was the inspiration for) our [`generate_cases`](https://github.com/dimagi/commcare-hq/blob/0a2e8a379196a56fbabdafcfcb922cfce093a1a9/corehq/util/test_utils.py#L316) test utility.